### ISSUE TITLE
Broken link of ¨Edit sample detail¨ on My samples page - eBMS website

### ISSUE THREAD

**Issue by CrisSevilleja** [BiologicalRecordsCentre/ABLE#667]
The small button of ¨Edit sample detail¨ on My samples page https://butterfly-monitoring.net/mydata/samples is broken bringing to the home page. 

![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/25052478/c789db71-27b6-41fb-be08-223d2eed3f4d)

**Comment by DavidRoy** [comment]
@andrewvanbreda - can you look at this please

**Comment by CrisSevilleja** [comment]
I think the link is broken for transect surveys of 2024, in previous surveys the link is working.

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja @DavidRoy 
Looking at this, I actually don't think it is even a new bug, it might be that circumstances have meant it is only getting noticed.
It looks like it is indeed dependent on the sample, but I think it is more complex than the age of the sample.

I am looking at the data, and I am left wondering this.
It MIGHT be that samples created on an app are the ones not working.

What do you think?

The reason I think this is that it is samples without an External Key are the ones not working.
The grid tries to display the editing page based on the data entry page that is saved into a field on the sample.
However if it is a sample from an app, it won't fill in that database field.

If this is the case, this does leave a problem on how to get this to work.
It might be necessary to put the EBMS 15 minute counts onto a separate grid.

Would you be able to confirm if the failing ones are created on app or website?

Thanks

Andy

**Comment by DavidRoy** [comment]
@andrewvanbreda these are app surveys I think as 15 minute counts are not possible via the website

**Comment by andrewvanbreda** [comment]
@DavidRoy 
Interesting. As the 15 minute count ones work. I notice they do have external key.

The problem is only some of the EBMS Transects ones.
Sample 24371039 works, but 25112615 doesn't.
Both of the those are EBMS transects.
I suspect that 24371039 was created on the website, but 25112615 was entered on app.
I do not know about the app, so you will need to confirm if the app even allows transect sample entry.

**Comment by DavidRoy** [comment]
if it's transects, it could be either the website or the app.
@kazlauskis to comment on the external key for transect entry via the app

**Comment by andrewvanbreda** [comment]
Hi @kazlauskis 
The external_key isn't what is breaking it, but that is my clue that they might be app samples. 
What might break it is if the input_form field isn't filled in on the sample..or I should say it would probably break it with the way My Samples on the website currently works.
Do you have access to the database (I don't at the moment)? I would be interested to know if the app is creating samples with the input_form field filled in as walk-data-entry.
If it isn't, whether it is possible to add that, or whether I need to change the way the My Samples page works to get this working.
Thanks

**Comment by andrewvanbreda** [comment]
@kazlauskis @DavidRoy @CrisSevilleja 
Actually Karolis, do not worry, I have noticed the website is calling some Javascript when the user clicks to edit a sample.
I have changed this to always force the system to go to the walk-data-entry page when survey 562 is detected.
The samples grid is now fixed on live.
Leave this issue open for now and I will fix the Occurrences grid tomorrow morning as need a little more work.

**Comment by CrisSevilleja** [comment]
Hi @andrewvanbreda then you found what the problem was, I can confirm the broken link is only for the transects registered with the app. Transects registered on the website and 15min counts done in the app have the Edit sample button working. 

Examples of my transects registered with the app: https://butterfly-monitoring.net/undefined?sample_id=22048619, https://butterfly-monitoring.net/undefined?sample_id=25112615
Thanks for looking at this so fast,

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja @DavidRoy
Fixed.
The Sample/Occurrences grids on My Samples no longer use the input_form from the samples database table.
Instead both look at the survey ID instead, then force the opening of the correct data entry page based on that.

**Comment by andrewvanbreda** [comment]
Close if happy

**Comment by DavidRoy** [comment]
@CrisSevilleja please close if happy

**Comment by CrisSevilleja** [comment]
For the users who reported this issue, the edit link is working. For me, it is not working for the transect survey https://butterfly-monitoring.net/undefined?sample_id=25112615 but for the rest yes, so I am fine with that. Maybe the website needs time to get the improvement running. 

thanks for solving it @andrewvanbreda

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja,
I just tried logging in as your user account, and I was able to open the edit screen for 25112615 ok.
If you continue to have a problem, perhaps try clearing the caches in your web browser.
Hope that helps.

**Comment by CrisSevilleja** [comment]
Hi again, it looks like this issue is also happening to moth trap visits as well and those are also coming from the app.

One user reported that survey ID 25243877 (EBMS fixed moth trap, 08/04/2024) cannot be edited under My samples page by clicking on the Edit sample button. Even, when entering on view and then Edit sample, he got an error message 
![WhatsApp Image 2024-04-10 at 11 14 24](https://github.com/BiologicalRecordsCentre/ABLE/assets/25052478/8a1f3eab-d276-4691-9674-5b9641aee4ec).
Anyway, the sample can be edited through My moth trap samples https://butterfly-monitoring.net/my-moth-trap-samples but not through My samples...

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja
I have fixed the specific problem you had there by using the same technique as before, the moth trap editing page will now open.

The moth trap samples page was fine because all the samples come from the same survey, so it doesn't even need to check the sample to open the right page.

I think there might still be a problem with the way this is working though.
Would you be able to check if you and your colleagues have the issue I have noticed please?
The possible problem is this:
If you are using a language that is not English, when you click to edit a sample on My Samples, is it switching to English? or is the edit sample page opened in the correct language?
Thanks for your help.

**Comment by CrisSevilleja** [comment]
Hi @andrewvanbreda 
thanks for fixing this issue, now it is working well. 

Regarding your question, I tried, and for a Spanish and a Danish user that problem doesn't happen. when I select "edit a sample" under My sample page, the selected languages, Spanish and Danish stay perfectly, not switching to English. However, when I tried to do it on my account (English as the selected language), if I selected another language (Romanian in this case), then I clicked on "edit a sample" under My sample page, for my account the loaded page switched to English. This issue we know, discussed previously with @JimBacon, as I have English as selected language, the website normally comes back to English when loading a new page. 

I hope this is useful.
